raft/src/test/java/org/apache/kafka/raft/RaftEventSimulationTest.java

```diff
@@ -1326,42 +1745,47 @@ private void assertCommittedData(RaftNode node) {
                 return;
             }
 
-            AtomicLong startOffset = new AtomicLong(0);
-            log.earliestSnapshotId().ifPresent(snapshotId -> {
-                assertTrue(snapshotId.offset() <= highWatermark.getAsLong());
-                startOffset.set(snapshotId.offset());
+            AtomicLong startOffset = new AtomicLong(log.startOffset());
 
-                try (SnapshotReader<Integer> snapshot = RecordsSnapshotReader.of(
+            // We do not perform this check with KIP-853 enabled
+            // because we write the bootstrap snapshot to all initial voters
+            if (!cluster.withKip853) {
```

Comment:
This is not because of KIP-853. This is because the bootstrap checkpoint (`offset == 0`) doesn't contain any data batches. It only contains control batches.